diesel/src/macros/associations/belongs_to.rs

```diff
@@ -45,15 +45,25 @@
 ///
 /// [custom_derive]: https://crates.io/crates/custom_derive
 ///
-/// ```ignore
+/// ```no_run
+/// # #[macro_use] extern crate diesel;
+/// # #[macro_use] extern crate custom_derive;
+/// # table! { users { id -> Integer, } }
+/// # table! { posts { id -> Integer, user_id -> Integer, } }
+/// # custom_derive!{
+/// #     #[derive(Identifiable(users))]
+/// #     pub struct User {
+/// #         id: i32,
+/// #     }
+/// # }
 /// custom_derive! {
-///     #[derive(BelongsTo(User, foreign_key = user_id)]
-///     #[table_name(posts)]
+///     #[derive(BelongsTo(User, foreign_key = user_id, table = posts), Identifiable(posts))]
```

Comment:
In theory yes.
In practice custom_derive removes every attribute from the item, so `#[table_name(posts)]` is not passed to the underlying macro.

Secondly, the macro is passed the entire item, **sans attributes**. It is the derivation macro's job to parse the item correctly.

[Documentation](https://danielkeep.github.io/rust-custom-derive/doc/custom_derive/index.html) of custom_derive, below the example code.